Source/Core/DiscIO/RiivolutionPatcher.cpp

```diff
@@ -522,6 +523,11 @@ static bool MemoryMatchesAt(const Core::CPUThreadGuard& guard, u32 offset,
 static void ApplyMemoryPatch(const Core::CPUThreadGuard& guard, u32 offset,
                              const std::vector<u8>& value, const std::vector<u8>& original)
 {
+#ifdef USE_RETRO_ACHIEVEMENTS
+  if (::Config::Get(::Config::RA_HARDCORE_ENABLED))
+    return;
+#endif  // USE_RETRO_ACHIEVEMENTS
```

Comment:
This will break some mods. I would suggest detecting these patches as part of hashing the game, if possible.

Reply:
I plan on it but that would be a future update in a different PR and to my knowledge isn't a blocking requirement for getting this first round out.